test/extensions/filters/udp/udp_proxy/udp_proxy_filter_test.cc

```diff
@@ -1468,10 +1492,13 @@ class HttpUpstreamImplTest : public testing::Test {
                   absl::optional<std::string> opt_authority = absl::nullopt,
                   absl::optional<std::string> opt_path = absl::nullopt,
                   absl::optional<HeaderToAdd> header_to_add = absl::nullopt) {
+    // In case connect-udp is used, Envoy expect the H2 headers to be normalized with H1,
+    // so expect that the request headers here match H1 CONNECT headers, even though
+    // eventually H2 headers will be sent.
     std::string scheme = is_ssl ? "https" : "http";
     std::string authority =
         opt_authority.has_value() ? opt_authority.value() : "default.host.com:10";
-    std::string method = use_post ? "POST" : "CONNECT";
+    std::string method = use_post ? "POST" : "GET";
```

Comment:
The comment just above says to expect H1 CONNECT but here we use "GET". This is surprising. Can you elaborate?

Reply:
Done